File: packages/python-sdk/psti_sdk/resources/pastes.py
from typing import Dict, Any, Optional

class PastesResource:
    def __init__(self, client):
        self.client = client

    def create(self, title: str, content: str, language: str = "plaintext", 
               visibility: str = "public", encrypted: bool = False, 
               burn_after_read: bool = False, **kwargs) -> Dict[str, Any]:
        """Create a new paste."""
        payload = {
            "title": title,
            "content": content,
            "language": language,
            "visibility": visibility,
            "encrypted": encrypted,
            "burn_after_read": burn_after_read
        }
        payload.update(kwargs)
        return self.client.request("POST", "/pastes", json=payload)

    def get(self, paste_id: str, password: Optional[str] = None) -> Dict[str, Any]:
        """Get a paste by ID."""
        params = {"password": password} if password else None
        return self.client.request("GET", f"/pastes/{paste_id}", params=params)

    def list(self) -> Dict[str, Any]:
        """List pastes for the current user."""
        return self.client.request("GET", "/pastes")

    def fork(self, paste_id: str, password: Optional[str] = None) -> Dict[str, Any]:
        """Fork an existing paste."""
        payload = {"password": password} if password else {}
        return self.client.request("POST", f"/pastes/{paste_id}/fork", json=payload)
        
    def get_versions(self, paste_id: str, password: Optional[str] = None) -> Dict[str, Any]:
        """Get the version history of a paste by its ID."""
        params = {"password": password} if password else None
        return self.client.request("GET", f"/pastes/{paste_id}/versions", params=params)

    def update(self, paste_id: str, data: Dict[str, Any]) -> Dict[str, Any]:
        """Update a paste."""
        return self.client.request("PATCH", f"/pastes/{paste_id}", json=data)

    def delete(self, paste_id: str) -> Dict[str, Any]:
        """Delete a paste."""
        return self.client.request("DELETE", f"/pastes/{paste_id}")

File: packages/sdk/src/resources/pastes.ts
import { CreatePasteInput, UpdatePasteInput } from '@psti/validation';
import { Paste, PasteVersion, ApiResponse } from '@psti/types';
import { PstiClient } from '../client';

export class PastesResource {
    constructor(private readonly client: PstiClient) {}

    create(input: CreatePasteInput): Promise<ApiResponse<Paste>> {
        return this.client.request<Paste>('/api/v1/pastes', {
            method: 'POST',
            body: JSON.stringify(input),
        });
    }

    get(id: string, password?: string): Promise<ApiResponse<Paste>> {
        const queryParams = password ? `?password=${encodeURIComponent(password)}` : '';
        return this.client.request<Paste>(`/api/v1/pastes/${id}${queryParams}`, {
            method: 'GET',
        });
    }

    list(): Promise<ApiResponse<Paste[]>> {
        return this.client.request<Paste[]>('/api/v1/pastes', {
            method: 'GET',
        });
    }

    fork(id: string, password?: string): Promise<ApiResponse<Paste>> {
        return this.client.request<Paste>(`/api/v1/pastes/${id}/fork`, {
            method: 'POST',
            body: JSON.stringify({ password }),
        });
    }

    getVersions(id: string, password?: string): Promise<ApiResponse<PasteVersion[]>> {
        const queryParams = password ? `?password=${encodeURIComponent(password)}` : '';
        return this.client.request<PasteVersion[]>(`/api/v1/pastes/${id}/versions${queryParams}`, {
            method: 'GET',
        });
    }

    update(id: string, input: UpdatePasteInput): Promise<ApiResponse<Paste>> {
        return this.client.request<Paste>(`/api/v1/pastes/${id}`, {
            method: 'PATCH',
            body: JSON.stringify(input),
        });
    }

    delete(id: string): Promise<ApiResponse<{ success: true }>> {
        return this.client.request<{ success: true }>(`/api/v1/pastes/${id}`, {
            method: 'DELETE',
        });
    }

    trackView(id: string, data: { referrer?: string; language?: string; screen_resolution?: string; timezone?: string }): Promise<ApiResponse<{ success: true }>> {
        return this.client.request<{ success: true }>(`/api/v1/pastes/${id}/views`, {
            method: 'POST',
            body: JSON.stringify(data),
        });
    }

    getAnalytics(): Promise<ApiResponse<any>> {
        return this.client.request<any>('/api/v1/pastes/analytics/aggregate', {
            method: 'GET',
        });
    }
}
